lib/utils/ignore.ts

```diff
@@ -320,17 +336,7 @@ export async function filterFilesWithDockerignore(
 				if (path.basename(relPath) === '.dockerignore') {
 					dockerignoreFiles.push(fileInfo);
 				}
-				for (const dir of dockerignoreServiceDirs) {
-					if (relPath.startsWith(dir)) {
-						if (!ignoreByDir[dir].ignores(relPath.substring(dir.length))) {
-							filteredFileList.push(fileInfo);
-						}
-						return;
-					}
-				}
-				if (!ignoreByDir[root].ignores(relPath)) {
-					filteredFileList.push(fileInfo);
-				}
+				filteredFileList.push(fileInfo);
```

Comment:
The build is failing:
https://ci.product-os.io/builds/35125
```
...
expected vs. found diff:
{
    "src/file1.sh": {
        "fileSize": 12,
        "type": "file"
    }
}
...
expected { '.dockerignore': { fileSize: 12, type: 'file' },
  'Dockerfile-alt': { fileSize: 13, type: 'file' },
  'file2-crlf.sh': { fileSize: 14, type: 'file' } } to deeply equal { '.dockerignore': { fileSize: 12, type: 'file' },
  'file2-crlf.sh': { fileSize: 14, type: 'file' },
  'src/file1.sh': { fileSize: 12, type: 'file' },
  'Dockerfile-alt': { fileSize: 13, type: 'file' } }
```
I believe at this test case:
https://github.com/balena-io/balena-cli/blob/v12.51.1/tests/commands/build.spec.ts#L461
```ts
it('should create the expected tar stream (docker-compose, --multi-dockerignore)'
```

I think (just from reading, I didn't run/debug it to confirm) that the following test data `.dockerignore` file was not processed as it should:
https://github.com/balena-io/balena-cli/blob/v12.51.1/tests/test-data/projects/docker-compose/basic/service2/.dockerignore
```
**/file1.sh
```
Such that `src/file1.sh` was not ignored as it should have been.